gc/base/ParallelDispatcher.cpp

```diff
@@ -387,7 +393,20 @@ MM_ParallelDispatcher::shutDownThreads()
 void
 MM_ParallelDispatcher::wakeUpThreads(uintptr_t count)
 {
-	omrthread_monitor_notify_all(_slaveThreadMutex);	
+	/* This thread should notify and release _slaveThreadMutex asap. Threads waking up will need to
+	 * reacquire the mutex before proceeding with the task.
+	 *
+	 * Hybrid approach to notifying threads:
+	 * 	- Cheaper to do to individual notifies for small set of threads from a large pool
+	 * 	- More expensive to do with individual notifies with large set of threads
+	 */
+	if (count < OMR_MIN((_threadCountMaximum / 2), _extensions->dispatcherHybridNotifyThreadBound)) {
+		for (uintptr_t threads = 0; threads < count; threads++) {
+			omrthread_monitor_notify(_slaveThreadMutex);
```

Comment:
Was any consideration done with regards to doing a single notify here and then when a thread wakes up if more threads need to be notified it could do the notify at that point? That way each thread would only ever do a single notify.

Reply:
It was one of the approaches that I tried, the results were actually worse. At best they were the same as doing single notifies at this point. It seemed like a good approach, but it was abandoned it given the results, I didn't investigate any further.